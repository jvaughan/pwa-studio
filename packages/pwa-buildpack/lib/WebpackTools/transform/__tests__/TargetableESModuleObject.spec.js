const TargetableESModuleObject = require('../TargetableESModuleObject');
const ESMCollectionLoader = require('../../loaders/export-esm-collection-loader');
const testWebpackLoader = require('../../../TestHelpers/testWebpackLoader');

test('builds a module which exports an array of its imports', async () => {
    const moduleObject = new TargetableESModuleObject(
        './module-object.js',
        () => {}
    );
    moduleObject.set('import first from "first"', '{ second } from "second";');
    moduleObject.set('{ third as second } from "third"');
    moduleObject.set('{ fourth as third } from "fourth"');
    moduleObject.set('* as first from "elsewhere"');
    moduleObject.set('{ second as silver } from "second"');
    const transform = moduleObject.flush();
    expect(transform).toHaveLength(1);
    expect(transform[0]).toMatchSnapshot();
    const { output } = await testWebpackLoader.runLoader(
        ESMCollectionLoader,
        '/** this-un exports an object with mappings */\nexport default {};',
        {
            query: [transform[0].options]
        }
    );
    expect(output).toMatchSnapshot();
});
