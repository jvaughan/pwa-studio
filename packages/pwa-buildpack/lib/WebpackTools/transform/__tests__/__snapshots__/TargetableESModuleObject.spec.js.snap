// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builds a module which exports an array of its imports 1`] = `
Object {
  "fileToTransform": "./module-object.js",
  "options": Object {
    "items": Object {
      "import * as first from \\"elsewhere\\";": "first",
      "import { fourth as third } from \\"fourth\\";": "third",
      "import { second as silver } from \\"second\\";": "silver",
      "import { third as second } from \\"third\\";": "second",
    },
    "warnings": Array [
      "TargetableESModuleObject \\"./module-object.js\\": Export \\"first\\" was already assigned to \\"import first from \\"first\\";\\". Overwriting to \\"*\\" from \\"elsewhere\\".",
    ],
  },
  "transformModule": "@magento/pwa-buildpack/lib/WebpackTools/loaders/export-esm-collection-loader",
  "type": "source",
}
`;

exports[`builds a module which exports an array of its imports 2`] = `
"/** this-un exports an object with mappings */
// generated by export-esm-collection-loader
import { third as second } from \\"third\\";
import { fourth as third } from \\"fourth\\";
import * as first from \\"elsewhere\\";
import { second as silver } from \\"second\\";

export default {
  second,
  third,
  first,
  silver
};"
`;
